File: Examples/Java/SDK/src/main/java/com/aspose/imaging/cloud/examples/manipulate/CropImageWithFormatChange.java
package com.aspose.imaging.cloud.examples.manipulate;

import com.aspose.imaging.cloud.examples.Utils;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardCopyOption;

public class CropImageWithFormatChange {

    public static void main(String... args) throws IOException {
        String input = "Sample1.png";
        String output = "Sample2.jpg";
        Path inputFile = Utils.getPath(CropImageWithFormatChange.class, input);
        Path outputFile = Utils.getPath(CropImageWithFormatChange.class, output);

        Integer x = 30;
        Integer y = 40;
        Integer width = 100;
        Integer height = 100;
        String format = "jpg";
        String outPath = "";

        Utils.getStorageSdk().PutCreate(
                input,
                null,
                Utils.STORAGE,
                inputFile.toFile()
        );

        com.aspose.imaging.model.ResponseMessage apiResponse
                = Utils.getImagingSdk().GetCropImage(
                        input,
                        format,
                        x,
                        y,
                        width,
                        height,
                        outPath,
                        Utils.FOLDER,
                        Utils.STORAGE
                );

        Files.copy(apiResponse.getInputStream(), outputFile, StandardCopyOption.REPLACE_EXISTING);

    }
}

File: Examples/Node.js/REST/manipulate/CropImageWithFormatChange.js
var utils = require("../utils.js");
var fs = require('fs');

var method = 'PUT';
var request_url = 'storage/file/' + 'Sample1.png';


var input_file = 'sample1.png';
var inputPath = getPath(__filename, input_file);
var output_file = 'nodejsREST.output.jpg';
var outputPath = getPath(__filename, output_file);

var input_buffer = fs.readFileSync(inputPath);


UploadFileBinary(
    method,
    Sign(request_url),
    input_buffer,
    function (response) {
        if (response.Status == 'OK') {
            console.log(inputPath, 'has been uploaded');
        }

        var method = 'GET';
        var request_url = 'imaging/sample1.png/crop?format=jpeg&x=30&y=40&width=100&height=100&outPath=' + output_file;
        ProcessCommandContent(
            method,
            Sign(request_url),
            null,
            function (buffer) {
                fs.writeFileSync(outputPath, buffer);
                console.log('File Saved' , outputPath);

            }
        );
    }
);
